src/os/bluestore/Allocator.h

```diff
@@ -46,6 +46,18 @@ class Allocator {
   virtual void release(
     uint64_t offset, uint64_t length) = 0;
 
+  /* Bulk release. Implementations may override this method to handle the whole
+   * set at once. This could save e.g. unnecessary mutex dance. */
+  virtual void release(const interval_set<uint64_t>& release_set) {
+    /* TODO(rzarzynski): make this pure virtual and eradicate the single-op
+     * release() after switching all allocators. */
+    for (interval_set<uint64_t>::const_iterator p = release_set.begin();
+	 p != release_set.end();
+	 ++p) {
+      release(p.get_start(), p.get_len());
+    }
+  }
+
```

Comment:
it's probably the same amount of work to just implement this in bitmapallocator and leave this pure virtual in Allocator.

Reply:
Entirely true. :-) Just pushed a commit for it.